components/brave_vpn/browser/connection/win/utils.cc

```diff
@@ -42,21 +43,27 @@ class ScopedHeapAlloc {
 };
 
 // https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage
-void PrintSystemError(DWORD error) {
+std::string PrintSystemError(DWORD error) {
   constexpr DWORD kBufSize = 512;
   TCHAR lpsz_error_string[kBufSize];
 
   DWORD buf_len =
       FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
                     NULL, error, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
                     lpsz_error_string, kBufSize, NULL);
-  if (buf_len) {
-    LOG(ERROR) << lpsz_error_string;
-  }
+  if (!buf_len)
+    return "";
+
+  LOG(ERROR) << lpsz_error_string;
+  return base::WideToUTF8(lpsz_error_string);
 }
 
 // https://docs.microsoft.com/en-us/windows/win32/api/ras/nf-ras-rassetcredentialsa
-DWORD SetCredentials(LPCTSTR entry_name, LPCTSTR username, LPCTSTR password) {
+DWORD SetCredentials(LPCTSTR entry_name,
```

Comment:
I did not find where return DWORD code is used, just a chekc for boolean value bellow, I think it worh to use `absl::optional<std::string> SetCredentials` to return error or absl::nullopt if success